src/vs/workbench/parts/files/electron-browser/fileActions.ts

```diff
@@ -1335,35 +1336,77 @@ export function validateFileName(parent: IFileStat, name: string, allowOverwriti
 		return nls.localize('emptyFileNameError', "A file or folder name must be provided.");
 	}
 
+	const names: string[] = trimTrailingSlashes(name)	// prevents empty last array element after split
+		.split(/[\\/]/);
+
+
 	// Do not allow to overwrite existing file
 	if (!allowOverwriting) {
-		if (parent.children && parent.children.some((c) => {
-			if (isLinux) {
-				return c.name === name;
+		let p = parent;
+
+		const alreadyExisting = names.every((folderName) => {
+			let { exists, child } = alreadyExists(p, folderName);
+
+			if (!exists) {
+				return false;
+			} else {
+				p = child;
+				return true;
 			}
+		});
 
-			return c.name.toLowerCase() === name.toLowerCase();
-		})) {
+		if (alreadyExisting) {
 			return nls.localize('fileNameExistsError', "A file or folder **{0}** already exists at this location. Please choose a different name.", name);
 		}
 	}
 
 	// Invalid File name
-	if (!paths.isValidBasename(name)) {
+	if (names.some((folderName) => !paths.isValidBasename(folderName))) {
 		return nls.localize('invalidFileNameError', "The name **{0}** is not valid as a file or folder name. Please choose a different name.", trimLongName(name));
 	}
 
-	// Max length restriction (on Windows)
 	if (isWindows) {
-		const fullPathLength = name.length + parent.resource.fsPath.length + 1 /* path segment */;
-		if (fullPathLength > 255) {
+		if (windowsMaxLengthRestrictionReached(name, parent)) {
 			return nls.localize('filePathTooLongError', "The name **{0}** results in a path that is too long. Please choose a shorter name.", trimLongName(name));
 		}
 	}
 
 	return null;
 }
 
+function windowsMaxLengthRestrictionReached(name: string, parent: IFileStat): boolean {
```

Comment:
I do not understand what is hte benefit of adding this function. It is the same functionality as before but now it is enapsulated in a function, but we still call the function only once.
Imho a best PR is when it contains the minimal changes need to make it happen, so I would remove this helper function